File: hm.py
import redis


def test_redis_connection():
    try:
        # Connect to Redis
        client = redis.StrictRedis(host='localhost', port=6379, db=0)

        # Set a key-value pair in Redis
        client.set('testKey', 'Hello from Python!')

        # Retrieve the value for the key
        value = client.get('testKey')

        # Print the retrieved value
        if value:
            print(f"testKey: {value.decode('utf-8')}")
        else:
            print("Could not retrieve the value for 'testKey'.")

    except Exception as e:
        print(f"Error connecting to Redis: {e}")


if __name__ == "__main__":
    test_redis_connection()

File: hm.js
import redis from 'redis'

// Create a Redis client
const client = redis.createClient({
  url: 'redis://localhost:6379'  // Modify the URL if needed (for example, use the container IP)
});

(async () => {
  try {
    // Connect to Redis
    await client.connect();
    console.log("Connected to Redis!");

    // Set a key-value pair
    await client.set('testKey', 'Hello from Codespaces!');

    // Get the value of the key
    const value = await client.get('testKey');
    console.log('testKey:', value);

    // Disconnect from Redis
    await client.quit();
    console.log("Disconnected from Redis!");
  } catch (error) {
    console.error("Error connecting to Redis:", error);
  }
})();
